lib/restaurant.rb

```diff
@@ -0,0 +1,20 @@
+
+class Restaurant
+  def initialize( messenger, menu)
+    @messenger = messenger
+    @menu = menu
+    @header = { from: '+441290211165', to: '+447527994300'}
+  end
+
+  def request_takeaway
+    takeaway = Takeaway.new(self, @menu)
```

Comment:
this completely couples Takeaway and Restaurant - you cannot have one without the other. This makes it harder to reuse or maintain your code. Use DI instead.